### PR TITLE
Force assertions enabled for all test case builds

### PR DESCRIPTION
**Pull request by tyrantlizard77**
This is done by sanitycheck, but not manual builds and it keeps biting us, especially when testing on real hardware. Force assertions on for all tests, disabling them for benchmarks.

### CODE REVIEW

File: tests/benchmarks/timing_info/prj.conf [hunk 1]
```diff
@@ -2,4 +2,4 @@ CONFIG_TEST=y
 CONFIG_EXECUTION_BENCHMARKING=y
 CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_HEAP_MEM_POOL_SIZE=256
-CONFIG_MAIN_STACK_SIZE=2048
\ No newline at end of file
+CONFIG_MAIN_STACK_SIZE=2048CONFIG_FORCE_NO_ASSERT=y
```
Comment (nashif) [inline tests/benchmarks/timing_info/prj.conf]: hit by a newline disaster

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by andyross — APPROVED** [review]
Seems sane.  Might be worth patching sanitycheck to remove/deprecate the --enable/disable-asserts option as it's now mostly duplicating what the tests tree already does.

**Comment by andrewboie** [comment]
@andyross good idea

**Review by nashif — APPROVED** [review]

File: subsys/debug/Kconfig [hunk 2]
```diff
@@ -84,7 +84,7 @@ config EARLY_CONSOLE
 config ASSERT
 	bool
 	prompt "Enable __ASSERT() macro"
-	default n
+	default y if TEST
```
Comment (ulfalizer) [inline subsys/debug/Kconfig]: Just `default TEST` should work here too. Might be inconsistent with the style elsewhere though.

**Review by ulfalizer — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by marc-hb** [comment]
> Might be worth patching sanitycheck to **remove/deprecate** the --enable/disable-asserts option

Deprecating is warning the users that something will be removed in the future but commit 29599f6d80bb made the options no-ops immediately.

Assuming asserts are on by default, I think `--enable-asserts` should print a "NO-OP kept for backward compatibility" warning while `--disable-asserts` is harmful (it just hit me) hence it should be removed.

Or maybe both can be just removed by now, it's been almost a year?

**Pull request merged**